src/mp_game_settings.cpp

```diff
@@ -171,23 +182,33 @@ void mp_game_settings::update_addon_requirements(const config & cfg) {
 
 	mp_game_settings::addon_version_info new_data(cfg);
 
+	std::map<std::string, addon_version_info>::iterator it = addons.find(cfg["id"].str());
 	// Check if this add-on already has an entry as a dependency for this scenario. If so, try to reconcile their version info,
 	// by taking the larger of the min versions. The version should be the same for all WML from the same add-on...
-	std::map<std::string, addon_version_info>::iterator it = addons.find(cfg["id"].str());
 	if (it != addons.end()) {
-		addon_version_info & addon = it->second;
+		addon_version_info& addon = it->second;
+
+		// an add-on can contain multiple types of content
+		// for example, an era and a scenario
+		for(const auto& item : new_data.content) {
+			addon.content.emplace_back(addon_content{ item.id, item.type });
+		}
+
+		if(new_data.required) {
+			addon.required = true;
```

Comment:
Hmm i think that if addon.required was already false that then the old value of  min_version should be ignored below scine we don't care for min_version for contents that don't require download

Reply:
If `new_data.required` is false, then no min_version would have been set for that add-on at that point.

Reply:
> If addon.required was false prior to the current add-on content being processed, then there wouldn't have been a min_version set.

I _think_ this is correct but i'm not sure, can you prove it?

Reply:
`addon.required` would only have already been set to true if a previous content from the same add-on had `new_data.required` set to true.  So if `addon.required` is false, that would mean that either:
a) This is the first content being processed for this add-on, so there wouldn't have been anything prior to set the min_version.
b) No previous content from this add-on passed the `if(new_data.required)` check, in which case the logic to set the min_version would have been skipped.

Reply:
If  `addon.required` is false then addon exists by assumption, if we for example added this addon for the second time, then addon is exactly the paremter of update_addon_requirements from when a content from that addon was added for the first time and the `addons.emplace(cfg["id"].str(), new_data` lines alone doesn't ensure that If addon.required was false prior to the current add-on content being processed, then there wouldn't have been a min_version set.

Reply:
So what's needed in the else block then is:
```
if(!new_data.required) {
	new_data.min_version = boost::none;
}
addons.emplace(cfg["id"].str(), new_data);
```
?

Reply:
i think that would work

Reply:
Updated.